### PR TITLE
SDIT 1992 Ignore 422 HTTP response when migrating a visit

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/helpers/WebClientHelpers.kt [hunk 1]
```diff
@@ -16,13 +16,11 @@ suspend inline fun <reified T : Any> WebClient.ResponseSpec.awaitBodyOrNullWhenN
     .onErrorResume(WebClientResponseException.NotAcceptable::class.java) { Mono.empty() }
     .awaitSingleOrNull()
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/helpers/WebClientHelpers.kt:18]: the 409 helper was never used so converted it to a 422 helper instead

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/helpers/WebClientHelpers.kt [hunk 2]
```diff
@@ -16,13 +16,11 @@ suspend inline fun <reified T : Any> WebClient.ResponseSpec.awaitBodyOrNullWhenN
...
 
-suspend inline fun <reified T : Any> WebClient.ResponseSpec.awaitBodyOrNullWhenConflict(): T? =
+suspend inline fun <reified T : Any> WebClient.ResponseSpec.awaitBodyOrNullWhenUnprocessableEntity(): T? =
   this.bodyToMono<T>()
-    .onErrorResume(WebClientResponseException.Conflict::class.java) { Mono.empty() }
+    .onErrorResume(WebClientResponseException.UnprocessableEntity::class.java) { Mono.empty() }
     .awaitSingleOrNull()
 
-class DuplicateMappingException(val error: DuplicateErrorResponse) : RuntimeException("message")
-
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/helpers/WebClientHelpers.kt:25]: not used

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/visits/VisitsMigrationService.kt [hunk 3]
```diff
@@ -74,37 +72,54 @@ class VisitsMigrationService(
       }
       ?: run {
         val nomisVisit = nomisApiService.getVisit(context.body.visitId)
-        determineRoomMapping(nomisVisit)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/visits/VisitsMigrationService.kt:77]: refactored to not use null as the control flow but response object instead to be explicit - it was getting very confusing

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**